### ISSUE TITLE
Neaby pin does not disappear after submitting to Wikidata, leading user to submit to Wikidata again

### ISSUE THREAD

**Issue by nicolas-raoul** [commons-app/apps-android-commons#2429]
**Steps to reproduce:** 

1. Go to Nearby
2. Select a pin near you
3. Go take a picture of it and upload it
4. Problem: The pin is still present of the map, maybe because the map is not refreshed or maybe because of some cache.
5. User thinks it did not work and tries again

Example where that happened: https://www.wikidata.org/w/index.php?title=Q5942564&action=history

By the community rules, a Wikidata item must have only one P18 image, not several (even if adding several is technically possible).
So the current app's behavior, in addition to being frustrating for the app user, can lead the user to unknowingly make edits that are frowned upon by the Wikidata community.

**Comment by VojtechDostal** [comment]
The database output often lags behind the edits by several minutes or even an hour. We cannot speed it up so we'd either have to remove the pin "artificially" from the app or entirely change the way pins are loaded.

**Comment by nicolas-raoul** [comment]
@VojtechDostal In my experience, simply killing the app and relaunching it makes the pin disappear, though. I will have to try a few times more, in a quick way :-) Having a filter of recently added pins sounds like a very feasible solution, it does not even have to be stored in database I guess.

**Comment by nicolas-raoul** [comment]
@VojtechDostal Is this issue the cause of the double P18 at https://www.wikidata.org/wiki/Q33323401 ?

**Comment by VojtechDostal** [comment]
@nicolas-raoul Yep, sorry for doing that intentionally :) . I wanted to find out if this is possible and then I forgot about it. Both pics are valid for the object but we might only keep one in the item, right?

**Comment by nicolas-raoul** [comment]
@VojtechDostal Yes, please choose the most representative :-)
